buildSrc/src/main/groovy/BuildkitBuild.groovy

```diff
@@ -0,0 +1,357 @@
+    private def gitTask
+    private def gitLogTask
+    private def releaseType
+
+    @Inject
+    protected abstract ObjectFactory getObjects()
+
+    @Inject
+    protected abstract ProviderFactory getProviderFactory()
+
+    // Credentials used for authentication to docker and artifactory
+    @Input
+    @Optional
+    final Property<String> registryUsername = getObjects().property(String).
+            convention(getProviderFactory().environmentVariable("CORDA_ARTIFACTORY_USERNAME")
+                    .orElse(getProviderFactory().gradleProperty("cordaArtifactoryUsername"))
+                    .orElse(getProviderFactory().systemProperty("corda.artifactory.username"))
+            )
+
+    @Input
+    @Optional
+    final Property<String> registryPassword = getObjects().property(String).
+            convention(getProviderFactory().environmentVariable("CORDA_ARTIFACTORY_PASSWORD")
+                    .orElse(getProviderFactory().gradleProperty("cordaArtifactoryPassword"))
+                    .orElse(getProviderFactory().systemProperty("corda.artifactory.password"))
+            )
+
+    // Property to set images for release 
+    @Input
+    final Property<Boolean> releaseCandidate =
+            getObjects().property(Boolean).convention(false)
+
+    //Property for nightly builds
+    @Input
+    final Property<Boolean> nightlyBuild =
+            getObjects().property(Boolean).convention(false)
+
+    // Property for pre-test
+    @Input
+    final Property<Boolean> preTest =
+            getObjects().property(Boolean).convention(false)
+
+    // Property to set the build tool used to create an image
+    // Currently supported tools are docker buildx and native builkit 
+    // NOTE: native buidkit requires port forwarding to the aws buildkit cluster
+    @Input
+    final Property<Boolean> isBuildx =
+            getObjects().property(Boolean).convention(true)
+
+    // Property for loading images into docker
+    @Input
+    final Property<Boolean> useDocker =
+            getObjects().property(Boolean).convention(false)
+
+    // The images contain shortened filenames if true
+    @Input
+    final Property<Boolean> useShortName =
+            getObjects().property(Boolean).convention(false)
+
+    // Property to set the output image's repository
+    @Input
+    final Property<String> containerRepo =
+            getObjects().property(String).convention('')
+
+    // Property used to set custom image tag
+    @Input
+    final Property<String> containerTag =
+            getObjects().property(String).convention('')
+
+    // Property used to set custom image name
+    @Input
+    final Property<String> containerName =
+            getObjects().property(String).convention('')
+
+    // Property to set the base image's name
+    @Input
+    final Property<String> baseImageName =
+            getObjects().property(String).convention('azul/zulu-openjdk')
+    // Property to set the base image's tag
```

Comment:
add newline before.

Also - I think it's more readable to have 1 line assignments for the properties here ;)